src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -139,6 +134,31 @@ RobotNavigationObstacleFactory::createObstaclesFromMotionConstraint(
     return obstacles;
 }
 
+std::vector<ObstaclePtr> RobotNavigationObstacleFactory::createGoalObstacles(
+    const Rectangle &goal) const
+{
+    // Adjust the goal obstacle radius slightly
+    const double goal_obstacle_radius = ROBOT_MAX_RADIUS_METERS - 0.01;
+
+    std::vector<ObstaclePtr> obstacles;
+
+    // Create top goal post obstacle
+    obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(Stadium(
+        Segment(goal.posXPosYCorner(), goal.negXPosYCorner()), goal_obstacle_radius)));
+
+    // Create bottom goal post obstacle
+    obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(Stadium(
+        Segment(goal.posXNegYCorner(), goal.negXNegYCorner()), goal_obstacle_radius)));
+
+    // Create left goal wall obstacle
+    obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(Stadium(
+        Segment(goal.negXPosYCorner(), goal.negXNegYCorner()), goal_obstacle_radius)));
```

Comment:
unfortunately, this specific obstacle doesn't work in both situations. In the friendly net, the back wall is from `negXPosYCorner` to `negXNegYCorner` but for the enemy net, the back wall is from `posXPosYCorner` to `posXNegYCorner`. It might be better to pass in the coordinates of the back wall directly to this function and create this line segment. The other obstacles seems fine.

Reply:
I will fix that, thank you!